client/src/components/CreateNewBoardModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import avatar from '../icons/avatar.jpg';
+import background from '../icons/teamBackground.jpg';
+import { DialogContext, ModalTypes } from '../contexts/DialogContext';
+
+const CreateNewBoardModal = () => {
+    const context = useContext(DialogContext);
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const onClickOutside = (e: any) => {
+        const element = e.target;
+        if (modalRef.current && !modalRef.current.contains(element)) {
+            e.preventDefault();
+            e.stopPropagation();
+            context.closeModalByType();
```

Comment:
I think an argument is missing in this function. A modal type is needed so that the DialogContext knows what modal should be closed.